core/vm/contracts.go

```diff
@@ -896,6 +897,40 @@ func (c *blake2F) Run(input []byte, caller common.Address, evm *EVM, gas uint64)
 	return output, gas, nil
 }
 
+// ed25519Verify implements a native Ed25519 signature verification.
+type ed25519Verify struct{}
+
+// RequiredGas returns the gas required to execute the pre-compiled contract.
+func (c *ed25519Verify) RequiredGas(input []byte) uint64 {
+	const sha2_512WordLength = 64
+
+	// round up to next whole word
+	lengthCeil := len(input) + sha2_512WordLength - 1
+	words := uint64(lengthCeil / sha2_512WordLength)
+	return params.Ed25519VerifyGas + params.Sha2_512BaseGas + (words * params.Sha2_512PerWordGas)
+}
+
+func (c *ed25519Verify) Run(input []byte, caller common.Address, evm *EVM, gas uint64) ([]byte, uint64, error) {
+	// Setup success/failure return values
```

Comment:
It looks like other precompiles are debiting gas while this precompile is not. Example from `blake2F`
```
	gas, err := debitRequiredGas(c, input, gas)
	if err != nil {
		return nil, gas, err
	}
```

Reply:
@trianglesphere thanks, somewhat missed that.

Updated the repo accordingly